### PR TITLE
vscode extension: migrate from any to unknown where possible

### PR DESCRIPTION
**Pull request by Veetaha**
`unknown` type is the stricter version of `any` and it should always be prefered (like `const` over `let`).
It lets you assign any value to it, but doesn't let you carry out arbitrary operations on them without an explicit type check (like `typeof unknownValue === 'string'`).

### CODE REVIEW

File: editors/code/src/commands/syntax_tree.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export function syntaxTree(ctx: Ctx): Cmd {
 
 // We need to order this after LS updates, but there's no API for that.
 // Hence, good old setTimeout.
-function afterLs(f: () => any) {
+function afterLs(f: () => unknown) {
```
Comment (matklad) [inline editors/code/src/commands/syntax_tree.ts]: Hm, this maybe should be `=> void`, or whats the TypeScript alternative for `-> ()`?

Comment (matklad) [reply editors/code/src/commands/syntax_tree.ts]: Also, this is just a horrible hack which needs to die in flames...

File: editors/code/src/ctx.ts [hunk 2]
```diff
@@ -73,11 +73,11 @@ export class Ctx {
         }
     }
 
-    get subscriptions(): { dispose(): any }[] {
+    get subscriptions(): { dispose(): unknown }[] {
         return this.extCtx.subscriptions;
     }
 
-    pushCleanup(d: { dispose(): any }) {
+    pushCleanup(d: { dispose(): unknown }) {
```
Comment (matklad) [inline editors/code/src/ctx.ts]: Similarly, this wants to be a `: void`?

**Comment by matklad** [comment]
bors d+

**Comment by Veetaha** [comment]
@matklad, I've noticed, that we don't invoke type checking at CI. I am not familiar with `rollup` at all (it is not popular bundler, the most popular is [`webpack`](https://github.com/webpack/webpack)), but it seems that it does not leverage the TypeScript type checker while bundling.

**Comment by Veetaha** [comment]
The easiest fix for that would be to do `npx tsc --noEmit`

**Comment by matklad** [comment]
Hm, yeah, we should make sure that rollup typechecks stuff

File: editors/code/src/ctx.ts (hunk 2 above)
Comment (Veetaha) [reply editors/code/src/ctx.ts]: In general, yes you are right, now when I am looking at it, this seems reasonable

File: editors/code/src/commands/syntax_tree.ts (hunk 1 above)
Comment (Veetaha) [reply editors/code/src/commands/syntax_tree.ts]: Okay, let me revisit this)

Comment (Veetaha) [reply editors/code/src/commands/syntax_tree.ts]: Yes, you are right, `void` is semantically `-> ()`, but not the unit type itself `()`. It is quite a special language item that is meant to be used only for the function return type and denotes that the value should be discarded (also function with `void` return type is covariant to other functions with different return type, in which way `void` type is different from `undefined` type). 

I would say that the concept of unit type the way it is present in Rust is not implemented in TypeScripts type system.

**Pull request merged**